src/scheduler/node_info.c

```diff
@@ -2956,6 +2956,14 @@ eval_simple_selspec(status *policy, chunk *chk, node_info **pninfo_arr,
 		return 0;
 	}
 
+	if (resresv->aoename != NULL) {
+		aoereq = find_resource_req(specreq_noncons, getallres(RES_AOE));
+		if ((aoereq == NULL && !pl->pack))
+			resresv->is_prov_needed = 0;
+		else
+			resresv->is_prov_needed = 1;
+	}
```

Comment:
I think this if statement is slightly wrong.  It says you need to provision if aoereq != NULL && !pl->pack.  In a chunk where aoereq != NULL, you shouldn't need to provision.

Reply:
> It says you need to provision if aoereq != NULL && !pl->pack
it's slightly different, we want provisionable nodes to be selected for a sub chunk only if either aoereq != NULL or if some other subchunk has requested aoe and placement is set to pack.

Reply:
I think the if statement has one last case which you don't want.  It says if aoereq == NULL && !pack.  If either side of that && is false, you'll set prov_needed.  This means if aoereq != NULL && !pack, you'll set prov_needed.

Reply:
Thanks, I understood and made changes accordingly.